src/transformers/trainer.py

```diff
@@ -521,6 +521,9 @@ def __init__(
                 if isinstance(model, PreTrainedModel):
                     # Patch the model with liger kernels. Use the default kernel configurations.
                     _apply_liger_kernel_to_instance(model=model)
+                elif isinstance(model.base_model.model, PreTrainedModel):
```

Comment:
If the model is neither a `PretrainedModel`, nor a `PeftModel`, we would get an `AttributeError` here. To guard against that, the line could be changed to:
```suggestion
                elif hasattr(model.get_base_model) and isinstance(model.get_base_model(), PreTrainedModel):
```
Also, since not everyone knows this, let's add a comment that this is to check for PEFT models.